c_src/filewatch_inotify.c

```diff
@@ -149,9 +169,55 @@ static ErlDrvSSizeT call(
 
   fail_alloc:
   fail_decode:
+  fail_op_check:
     return -1;
 }
 
+/* Return type to Erlang:
+   {port(), [{watch_descriptor(), filename:basename()}]}
+*/
+static void send_output(ErlDrvData self_)
+{
+    struct instance *self = (struct instance *)self_;
+
+    size_t pos = 0;
+    ErlDrvTermData buf[(MAX_EVENTS * EVENT_TUPLE_LENGTH) + BUFFER_TAIL_SIZE];
+
+    buf[pos++] = ERL_DRV_PORT;
+    buf[pos++] = driver_mk_port(self->port);
+
+    struct message *msg;
+    for(uint i = 0; i < self->n_events; ++i) {
+        msg = &(self->msgs[i]);
+        buf[pos++] = ERL_DRV_INT;
+        buf[pos++] = msg->wd;
+        buf[pos++] = ERL_DRV_STRING;
+        buf[pos++] = (ErlDrvTermData) msg->name;
+        buf[pos++] = strlen(msg->name);
+        buf[pos++] = ERL_DRV_TUPLE;
+        buf[pos++] = BUFFER_TUPLE_SIZE;
+    }
+
+    /* Append values designating Erlang type */
+    buf[pos++] = ERL_DRV_NIL;
+    buf[pos++] = ERL_DRV_LIST;
+    buf[pos++] = self->n_events + 1; // +1 for nil
+    buf[pos++] = ERL_DRV_TUPLE;
+    buf[pos++] = BUFFER_TUPLE_SIZE;
+
+    /* Reset self */
```

Comment:
This comment doesn't add much and I believe the line itself is pretty clear as to what it does.